src/cdk-experimental/selection/selection.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, EventEmitter, OnInit, Input, ElementRef} from '@angular/core';
+import {SelectionChange, SelectionModel, compareFn} from '@angular/cdk/collections';
+import {coerceBooleanProperty, coerceNumberProperty} from '@angular/cdk/coercion';
+
+
+@Directive({
+  selector: '[cdkSelection]'
+})
+export class CdkSelection<T> implements OnInit {
+
+  /** List of all active selections. */
+  @Input('cdkSelection') selections: T[] = [];
+
+  /**
+   * Different selection strategies to apply.
+   * - Single: Only one item can be selected at a time.
+   * - Multiple: Multiple items can be selected.
+   * - Modifier Multiple: Multiple items can be selected using modifier keys such as ctrl or shift.
+   */
+  @Input('cdkSelectionMode') mode: 'single' | 'multiple' = 'single';
+
+  /** When multi select is enabled, require a key modifier to select multiples. */
+  @Input('cdkSelectRequireModifier') requireModifier: boolean = false;
+
+  /** Track by used for selection matching. */
+  @Input('cdkSelectionTrackBy') trackBy: (model: T) => any;
+
+  /** Whether the selections can be cleared after selection. */
+  @Input('cdkSelectionDeselectable')
+  get deselectable(): boolean { return this._deselectable; }
+  set deselectable(val) { this._deselectable = coerceBooleanProperty(val); }
+  private _deselectable: boolean = true;
+
+  /** The max number of selections that can be selected */
+  @Input('cdkSelectionMaxSelections')
+  get maxSelections(): number { return this._maxSelections; }
+  set maxSelections(val) {
+    if (val !== undefined && val !== null) {
+      this._maxSelections = coerceNumberProperty(val);
+    }
+  }
+  private _maxSelections: number;
+
+  /** Whether the control is disabled or not. */
+  @Input('cdkSelectionDisabled')
+  get disabled(): boolean { return this._disabled; }
+  set disabled(val) { this._disabled = coerceBooleanProperty(val); }
+  private _disabled: boolean;
+
+  /** Event fired when a selection/deselection occurs. */
+  selectionChange = new EventEmitter<SelectionChange<T>>();
+
+  /** The model backing of the component. */
+  _selectionModel: SelectionModel<T>;
+
+  /** Set of registered toggle components. */
+  _set: any[] = [];
```

Comment:
This doesn't look like it needs to be `any`

Also `_set` isn't a great name

Reply:
How about `_toggleDirectives`?

I made it `any` because It actually is `CdkSelectionToggle` but including this file in this directive would create a circular dependency. Do you have an idea on a better approach?